scheduling-api-graphql/scheduling-api-graphql-fetchers/src/main/java/org/ow2/proactive/scheduling/api/graphql/fetchers/converter/JobInputConverter.java

```diff
@@ -153,6 +156,42 @@
         }).filter(array -> array.length > 0).collect(Collectors.toList());
     }
 
+    private JobStatus stringToJobStatus(String stringJobStatus) {
+        JobStatus jobStatus = null;
+        switch (stringJobStatus) {
+            case "CANCELED":
+                jobStatus = JobStatus.CANCELED;
+                break;
+            case "FAILED":
+                jobStatus = JobStatus.FAILED;
+                break;
+            case "FINISHED":
+                jobStatus = JobStatus.FINISHED;
+                break;
+            case "IN_ERROR":
+                jobStatus = JobStatus.IN_ERROR;
+                break;
+            case "KILLED":
+                jobStatus = JobStatus.KILLED;
+                break;
+            case "PAUSED":
+                jobStatus = JobStatus.PAUSED;
+                break;
+            case "PENDING":
+                jobStatus = JobStatus.PENDING;
+                break;
+            case "RUNNING":
+                jobStatus = JobStatus.RUNNING;
+                break;
+            case "STALLED":
```

Comment:
Maybe you could use `Enum::valueOf(..)` method? Like: `JobStatus.valueOf(stringJobStatus)`

Reply:
thanks man ! I didn't remember this at all :)